src/uwtools/drivers/forecast.py

```diff
@@ -142,78 +136,53 @@ def resources(self, platform: dict) -> Mapping:
         """
         Parses the config and returns a formatted dictionary for the batch script.
         """
-        # Add required fields to platform.
-        # Currently supporting only slurm scheduler.
-        return {
-            "account": platform["account"],
-            "nodes": 1,
-            "queue": "batch",
-            "scheduler": self.job_scheduler,
-            "tasks_per_node": 1,
-            "walltime": "00:01:00",
-        }
 
-    def run(self) -> None:
+        return self._config["jobinfo"].update({
+            "account": self._expt_config["user"]["account"],
+            "scheduler": self._expt_config["platform"]["scheduler"],
+        })
```

Comment:
One of these I'm going stop shooting myself in the foot with this one!

Reply:
Also, now that I think about it, it's probably not nice for me to be modifying the input config in this sneaky way.

Reply:
Good call about the copy. We use `copy.deepcopy()` elsewhere, so maybe that'd be good here, too.

Now that I'm thinking of it, in `FV3Forecast` we're now doing `self._config = self._expt_config["forecast"]` and I wonder if we should also be deep-copying that. Because...

...in general, maybe we should think about hygiene for these config objects. Certainly if/when we have something like a `Distillery`, it could pass _new, unique_ objects along so that there's no chance that one part of the code will modify something that will impact another. If we ever need to use concurrency, that will become a crucial concern. We could consider something like https://pypi.org/project/frozendict/ to make the config `dict` immediately immutable after parsing and validation -- provided we can still get mutable copies or mutable subtrees to pass along. At least that would make visible any places where we were trying to mutate the config. This could be something to think about sooner than later, as rooting out mutations later will be much harder than putting guardrails in place early on.

As a (maybe) interesting side note: Data structures in functional languages like Clojure and Haskell are immutable. In Clojure, for example, when you "modify" a map (aka a `dict`), it never happens in-place like with Python's `.update()` -- it's a call that returns a new map, and that new map is a _copy_ of the input map, so the old one is unchanged and, for example, another thread with a reference to the old map will not see the changes your thread has made and cannot be affected by them. Even better, Clojure uses "structural sharing", where a "copy" uses as much as possible of the old data structure's in-memory bytes so that copies are fast and memory efficient. No such luck for Python, but maybe we could find some safe conventions using copies, or `frozendict`, etc.